Code/GraphMol/RGroupDecomposition/RGroupCore.cpp

```diff
@@ -816,4 +817,26 @@ int RCore::matchingIndexToCoreIndex(int matchingIndex) const {
                   "Matched atom missing core index");
   return atom->getProp<int>(RLABEL_CORE_INDEX);
 }
+
+// Create tautomer query for the matching mol on demand and cache for performance
+std::shared_ptr<TautomerQuery> RCore::getMatchingTautomerQuery() {
+  if (!checkedForTautomerQuery) {
+    try {
+      // Enumerate tautomers from a sanitized copy of the matching molecule
+      RWMol copy(*matchingMol);
```

Comment:
I'm assuming the reason for the copy is because we need to call updatePropertyCache?